spring-xd-dirt/src/main/java/org/springframework/xd/dirt/rest/XDController.java

```diff
@@ -62,14 +64,20 @@
  * @author Gunnar Hillert
  */
 
-public abstract class XDController<D extends BaseDefinition, A extends ResourceAssemblerSupport<D, R>, R extends NamedResource> {
+public abstract class XDController<D extends BaseDefinition, A extends
+		ResourceAssemblerSupport<D, R>, R extends NamedResource, I extends BaseInstance<D>> {
 
-	private ResourceDeployer<D> deployer;
+	private AbstractInstancePersistingZKDeployer<D, I> deployer;
```

Comment:
Ilaya and I discussed this object yesterday. It seems to me that `ResourceDeployer` is being used for both deployment and as a CRUD object for streams/jobs. It would make sense to me to remove all of the CRUD functionality (i.e. `findAll()`) from this object and instead have the controller issue these queries through the repository interface directly.

However we decided to defer this refactoring until after this PR is merged.